commit a840865f5b44784f852638880e2681f6252babc5 (refs/changes/53/5553/1)
Author: Rui Loura <rui@joyent.com>
Date:   2019-02-08T16:27:55+00:00 (8 months ago)
    
    TRITON-954 sdcadm post-setup common-external-nics should be rack aware

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index 085b019..77936a1 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -4187,8 +4187,6 @@ function checkMissingExternalNics(opts, cb) {
     var doadminui = true;
     var doimgapi = true;
 
-    var netexternal;
-
     function getInstance(svcname, callback) {
         sapi.listServices({ name: svcname }, onServices);
 
@@ -4235,25 +4233,6 @@ function checkMissingExternalNics(opts, cb) {
                 next();
             });
         },
-        // Grab the external network details.
-        function (_, next) {
-            var listOpts = { name: 'external' };
-            napi.listNetworks(listOpts, function (err, nets) {
-                if (err) {
-                    cb(new errors.SDCClientError(err, 'napi'));
-                    return;
-                }
-
-                if (!nets.length) {
-                    cb(new Error(
-                        'Couldn\'t find external network in NAPI'));
-                    return;
-                }
-
-                netexternal = nets[0];
-                next();
-            });
-        },
         // Check what NICS the imgapi and adminui zones currently have. Only do
         // work for those which do not yet have an external nic.
         function (_, next) {
@@ -4276,10 +4255,10 @@ function checkMissingExternalNics(opts, cb) {
                 for (var i = 0, nic; i < nics.length; i++) {
                     nic = nics[i];
                     if (nic.belongs_to_uuid === svcadminui.uuid &&
-                        nic.nic_tag === 'external') {
+                        netconfig.isNicExternal(nic) {
                         doadminui = false;
                     } else if (nic.belongs_to_uuid === svcimgapi.uuid &&
-                        nic.nic_tag === 'external') {
+                        netconfig.isNicExternal(nic) {
                         doimgapi = false;
                     }
                 }
@@ -4296,7 +4275,6 @@ function checkMissingExternalNics(opts, cb) {
                 doadminui: doadminui,
                 svcadminui: svcadminui,
                 svcimgapi: svcimgapi,
-                netexternal: netexternal
             });
         }
     });
@@ -4304,23 +4282,20 @@ function checkMissingExternalNics(opts, cb) {
 
 SdcAdm.prototype.setupCommonExternalNics = function
 setupCommonExternalNics(opts, cb) {
-    var self = this;
     assert.func(opts.progress, 'options.progress');
 
+    var self = this;
+    var doadminui = true;
+    var doimgapi = true;
     var progress = opts.progress;
-
     var svcadminui;
     var svcimgapi;
-    var doadminui = true;
-    var doimgapi = true;
-
-    var netexternal;
 
     function addExternaNicToZone(svcobj, subcb) {
         var addparams = {
             uuid: svcobj.uuid,
             networks: [
-                { 'uuid': netexternal.uuid, primary: true }
+                { 'name': 'external', primary: true }
             ]
         };
         self.vmapi.addNicsAndWait(addparams, function (err, job) {
@@ -4344,7 +4319,6 @@ setupCommonExternalNics(opts, cb) {
                 return;
             }
 
-
             self.sapi.updateService(svcArr[0].uuid, {
                 params: {
                     networks: [
@@ -4373,7 +4347,6 @@ setupCommonExternalNics(opts, cb) {
                 doadminui = res.doadminui;
                 svcadminui = res.svcadminui;
                 svcimgapi = res.svcimgapi;
-                netexternal = res.netexternal;
                 next();
             });
         },
